backend_new/src/utilities/listing-url-slug.ts

```diff
@@ -0,0 +1,30 @@
+import Listing from '../dtos/listings/listing.dto';
+
+/*
+    This maps a listing to its url slug
+    This is used by the public site front end
+  */
+export function listingUrlSlug(listing: Listing): string {
+  const { name, listingsBuildingAddress } = listing;
+  if (!listingsBuildingAddress) {
+    return listingUrlSlugHelper(name);
+  }
+  return listingUrlSlugHelper(
+    [
+      name,
+      listingsBuildingAddress.street,
+      listingsBuildingAddress.city,
+      listingsBuildingAddress.state,
+    ].join(' '),
+  );
+}
+
+export function listingUrlSlugHelper(input: string): string {
+  return (
+    (input || '').match(
+      /[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]+|[0-9]+/g,
```

Comment:
This is probably a good function to add some unit tests for. Or at least add a comment above this to explain what this regex does

Reply:
oops! I forgot to write tests for this one, added!

Reply:
To what @ludtkemorgan said, I think we should also have a comment explaining the intent of this regex with an example